### ISSUE TITLE
URL Redirect Layer

### ISSUE THREAD

**Issue by alexpaxton** [influxdata/ui#408]
**Proposal:**
We (as in the design team or anyone running UX experiments) need a means of linking to specific areas of the UI from external sources such as email campaigns.

**Current behavior:**
This is impossible because every authenticated URL contains `orgID`, cloud provider, and region that we can't stick that in email links

**Desired behavior:**
**Redirect with parameter handling.**

A URL we can link to that redirects to specific locations in the UI. This link should be provider, org, and region agnostic. For example we'd want to link to the Notebooks page from an email but not have to worry about `orgID` and whatnot. Ideally this is something that can be built once and unblock non-engineers going forward

**Alternatives considered:**
Tried doing this through Marketo but we'd still need to get `orgID`, `provider` and `region` from salesforce (which didn't work out). 

**Use case:**
This drives engagement and internal understanding of users

**Comment by drdelambre** [comment]
you want to link to `/dashboards` and have it go to `/orgs/{whatever}/dashboards`? got lost in the whatnot

**Comment by alexpaxton** [comment]
@drdelambre yeah pretty much. This could probably be broken down into smaller pieces, for example, handling dashboards first and then adding redirects for different parts of the UI as needed

**Comment by barbaranelson** [comment]
Note: if we redirect via cloud, make sure we don't drop parameters, as currently happens for cloud2.influxdata.com (which is why design team uses the usage page URL).

http://cloud2.influxdata.com/usage/?walkme=14-301843

**Comment by corriganjohn** [comment]
After talking with @121watts  and Marc Delagrammatikas (ecomm team) morning --- this seems to be a relatively simple addition to what we are currently doing in the cloud2.influxdata.com redirect. We should be able to add a new routing (eg '/append_route') that takes the current routing we already build (region/provider/cloud2 url / orgid) and allows it to accept a string value where we can append the desired route and parameters.
Should this be an ecommerce issue?

**Comment by barbaranelson** [comment]
If it is in quartz code, then yes, please file a separate issue in the quartz project and we can take care of it there.

**Comment by drdelambre** [comment]
i totally want to get rid of the /org/{orgid} prefix for all of our urls, which should also satisfy this. all we need to add is an org switcher, and a little bit of code that removes that prefix for any user with old links saved.

idk if arbitrary string execution / routing is a good move, usually it's not

**Comment by subirjolly** [comment]
Do we still need this issue?

**Comment by barbaranelson** [comment]
I think we do.  Marketing wants to be able to send out emails with a link to some sub-element of the UI, and today we can't do that without figuring out the correct org id to include in the URL for each recipient.

**Comment by appletreeisyellow** [comment]
Done with deepLinking project https://github.com/influxdata/ui/pull/3508
